### ISSUE TITLE
state-visibility rule gives incorrect output when using `fallback()`

### ISSUE THREAD

**Issue by outdoteth** [protofire/solhint#313]
To reproduce:

Test.sol
```
fallback() external payable {
}
```

.solhint.json
```
{
    "extends": "solhint:recommended",
    "rules": {
        "func-visibility": ["warn", { "ignoreConstructors": true }],
    }
}
```

output:

```
/Users/t/Test.sol
  4:5  warning  Explicitly mark visibility of state  state-visibility
```

Expected no warnings because visibility is given by `external` keyword. It might be that the issue is the parser can't handle the `fallback()` function because it doesn't have the `function` keyword prefix (just a guess).

**Comment by dbale-altoros** [comment]
This issue is fixed when upgrading to the last solidity parser version (0.14.5)
https://github.com/protofire/solhint/pull/378
